plugins/backend/backstage-plugin-argo-cd-backend/src/plugin.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { ArgoCDBuilder } from './service/ArgoCdBuilder';
+
+export const ArgoCDPlugin = createBackendPlugin({
+  pluginId: 'argocd',
+  register(env) {
+    env.registerInit({
+      deps: {
+        http: coreServices.httpRouter,
+        logger: coreServices.logger,
+        config: coreServices.rootConfig,
+      },
+      async init({ http, logger, config }) {
+        logger.info('ArgoCD plugin is initializing');
+        const builder: ArgoCDBuilder = ArgoCDBuilder.createBuilder({
+          logger,
+          config,
+        });
+        const { router } = await builder.build();
```

Comment:
@nuritizra Karl is having me take a look at this PR. Similar to the other suggestion are we able to use `createRouter` here instead of the builder to avoid code duplication?